tools/pylib/_boutpp_build/backend.py

```diff
@@ -0,0 +1,216 @@
+import os  # corelib
+import glob  # corelib
+import hashlib  # corelib
+import base64  # corelib
+import tempfile  # corelib
+import subprocess  # corelib
+
+try:
+    import packaging.tags  # packaging
+except:
+    packaging = None
+
+
+def run(cmd):
+    print(f"running `{cmd}`")
+    ret = os.system(cmd)
+    assert ret == 0, f"{cmd} failed with {ret}"
+
+
+useLocalVersion = True
+
+
+def getversion(_cache={}):
+    if "r" not in _cache:
+        try:
```

Comment:
Flip conditional:

```suggestion
    if "r" in _cache:
        return _cache["r"].strip()
```

then rest of function doesn't need to be indented.

Also, what is `r`? Is it really `version`?

Reply:
It is not have have the return in two spots ...

`r` is for `return`. Will change.